File: src/main/java/se/clouds/app/javanet/app/Controllers/FeatureController.java
package se.clouds.app.javanet.app.Controllers;

import java.util.Base64.Encoder;

import se.clouds.app.javanet.app.domain.feature.query.GetFeature;
import se.clouds.app.javanet.core.controller.AppController;
import se.clouds.app.javanet.core.controller.IActionResult;
import se.clouds.app.javanet.core.di.Di;
import se.clouds.app.javanet.core.mediator.MediatR;

public class FeatureController extends AppController
{
    private MediatR<IActionResult> mediatr = (MediatR<IActionResult>)Di.GetMediator();
    private static String routePath = "/feature";

    public FeatureController()
    {
        super(routePath);
    }

    public IActionResult Get()
    {
        var result = mediatr.SendRequest(new GetFeature());
        return result.orElseThrow();
    }
}

File: core/src/main/dotnet/app/controllers/FeatureController.cs
using System.Threading;
using System.Threading.Tasks;
using MediatR;
using Microsoft.AspNetCore.Mvc;
using Microsoft.Extensions.Logging;
using se.clouds.javanet.app.domain.feature;
using se.clouds.javanet.app.domain.feature.queries;
using se.clouds.javanet.app.domain.model;

namespace se.clouds.javanet.app.Controllers
{
    [ApiController]
    [Route("[controller]")]
    public class FeatureController : ControllerBase
    {
        private readonly IMediator _mediator;
        private readonly ILogger<FeatureController> _logger;
        public FeatureController(IMediator mediator, ILogger<FeatureController> logger)
        {
            _mediator = mediator;
            _logger = logger;
        }
        [HttpGet]
        public async Task<SharedFeature[]> GetAsync()
        {
            return await _mediator.Send(new GetFeature(), new CancellationToken());
        }
    }
}
